controller/iteration_blackbox_test.go

```diff
@@ -763,3 +765,208 @@ func assertChildIterationLinking(t *testing.T, target *app.Iteration) {
+		fxt := tf.NewTestFixture(t, rest.DB, tf.Iterations(1))
+		svc, ctrl := rest.UnSecuredController()
+		test.DeleteIterationUnauthorized(t, svc.Context, svc, ctrl, fxt.Iterations[0].ID)
+	})
+
+	rest.T().Run("success - update workitems for deleted iteration", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, rest.DB,
+			tf.Iterations(2, func(fxt *tf.TestFixture, idx int) error {
+				if idx == 1 {
+					fxt.Iterations[idx].MakeChildOf(*fxt.Iterations[0])
+				}
+				return nil
+			}),
+			tf.WorkItems(5, func(fxt *tf.TestFixture, idx int) error {
+				wi := fxt.WorkItems[idx]
+				wi.Fields[workitem.SystemIteration] = fxt.Iterations[1].ID.String()
+				return nil
+			}))
+		svc, ctrl := rest.SecuredControllerWithIdentity(fxt.Identities[0])
+		iterationToDelete := fxt.Iterations[1]
+		test.DeleteIterationOK(t, svc.Context, svc, ctrl, iterationToDelete.ID)
+		wis, err := rest.db.WorkItems().LoadByIteration(svc.Context, iterationToDelete.ID)
+		require.Nil(t, err)
+		assert.Empty(t, wis)
+	})
+
+	// Following test creates the strucute shown in diagram
+	// root Iteration
+	// |___________Iteration 1 (5 WI)
+	// |                |___________Iteration 2 (5 WI)
+	// |                                |___________Iteration 3 (5 WI)
+	// |___________Iteration 4 (2 WI)
+	//                     |___________Iteration 5 (3 WI)
+	// then deletes iteration1 & iteration5 to verify the effect
+	rest.T().Run("success - verify that workitems are updated correctly", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, rest.DB,
+			tf.Iterations(6,
+				func(fxt *tf.TestFixture, idx int) error {
+					i := fxt.Iterations[idx]
+					switch idx {
+					case 1:
+						i.MakeChildOf(*fxt.Iterations[0])
+					case 2:
+						i.MakeChildOf(*fxt.Iterations[1])
+					case 3:
+						i.MakeChildOf(*fxt.Iterations[2])
+					case 4:
+						i.MakeChildOf(*fxt.Iterations[0])
+					case 5:
+						i.MakeChildOf(*fxt.Iterations[4])
+					}
+					return nil
+				}),
+			tf.WorkItems(20, func(fxt *tf.TestFixture, idx int) error {
+				wi := fxt.WorkItems[idx]
+				switch idx {
+				case 0, 1, 2, 3, 4:
+					wi.Fields[workitem.SystemIteration] = fxt.Iterations[1].ID.String()
+				case 5, 6, 7, 8, 9:
+					wi.Fields[workitem.SystemIteration] = fxt.Iterations[2].ID.String()
+				case 10, 11, 12, 13, 14:
+					wi.Fields[workitem.SystemIteration] = fxt.Iterations[3].ID.String()
+				case 15, 16:
+					wi.Fields[workitem.SystemIteration] = fxt.Iterations[4].ID.String()
+				case 17, 18, 19:
+					wi.Fields[workitem.SystemIteration] = fxt.Iterations[5].ID.String()
+				}
+				return nil
+			}))
+		svc, ctrl := rest.SecuredControllerWithIdentity(fxt.Identities[0])
+		iterationToDelete := fxt.Iterations[1]
+		test.DeleteIterationOK(t, svc.Context, svc, ctrl, iterationToDelete.ID)
+		wis, err := rest.db.WorkItems().LoadByIteration(svc.Context, iterationToDelete.ID)
+		require.Nil(t, err)
+		assert.Empty(t, wis)
+
+		// Verify that 20 WIs are moved to Root iteration
```

Comment:
if you remove `iteration 1`, you should have 15 work items moved to the `root` iteration, shouldn't you ?

Reply:
Oh! yeah, the comment was wrong, test expects 15 only https://github.com/fabric8-services/fabric8-wit/pull/1683/files#diff-616d852d748d6065595c347aa3e715d7R946
thanks. done in https://github.com/fabric8-services/fabric8-wit/pull/1683/commits/21a922cabd1ddca24311e954ec69455efd52b824